### DISCUSSION TITLE
How to use firebase admin in getStaticProps?

### DISCUSSION THREAD

**Discussion by iammarkps** [vercel/next.js#11782] [Help]
I currently use getStaticProps to generate page from firebase firestore. And I don't want to add api calling overhead. As you can see in https://github.com/programming-in-th/programming.in.th/blob/b3999a501f2df53a876f66e2770fb7dad797c318/src/pages/tasks/%5Bslug%5D.tsx#L68 and https://github.com/programming-in-th/programming.in.th/blob/b3999a501f2df53a876f66e2770fb7dad797c318/src/pages/tasks/index.tsx#L234. It's fetching logic is in https://github.com/programming-in-th/cloud_functions/blob/master/functions/src/tasks.ts. It's use firebase admin to fetch from firestore. How can I initialize firebase admin only in getStaticProps?

**Comment by dheeraj-p** [comment]
Since it is static data(I assume because you are using getStaticProps). Why can't you just store the data in a JSON file and load it in your getStaticProps?

**Reply by iammarkps** [reply]
Many people can add data into the data i've used. So I think JSON won't be a very good idea for this. But the data update isn't frequent so I can use getStaticProps.

**Reply by Jagadesh-Ram** [reply]
any updates here? I want to getStaticProps() to initialize the firebase auth and fetch the data in build time itself!

**Comment by ingoclaro** [comment]
not sure what you mean by 
> How can I initialize firebase admin only in getStaticProps?

can you elaborate?

**Reply by iammarkps** [reply]
Yes, firebase admin need to be initialize only in secure env (eg. In server). But my app also use firebase client to handle some task. So I need to use firebase admin only in getStaticProps not all my app.

**Comment by JonasJW** [comment]
I have the same problem...

**Comment by julbrs** [comment]
I am in the same issue here. I have a very small base code and my goal is just to call firebase-admin from the getStaticProps method, it seems not easy at all.

For reference I am trying to check the following repo : https://docs.react2025.com/feedback/dynamic-routing https://github.com/leerob/fastfeedback it's calling firebase-admin directly from the getStaticProps.

Update: I use to have in my `next.config.js` the following:

```
module.exports = {
  // for netlify
  target: "serverless",
};

```

For netlify deploy (see [here](https://docs.netlify.com/configure-builds/common-configurations/next-js/)).

I am getting this error:

```
info  - Loaded env from /Users/julien/code/montessori/montessori-ressources-ui/.env.local
info  - Loaded env from /Users/julien/code/montessori/montessori-ressources-ui/.env
info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
info  - Checking validity of types  
info  - Creating an optimized production build  
Failed to compile.

ModuleNotFoundError: Module not found: Error: Can't resolve '@firebase/app' in '/Users/julien/code/montessori/montessori-ressources-ui/node_modules/@firebase/database/dist'


> Build error occurred
Error: > Build failed because of webpack errors
    at /Users/julien/code/montessori/montessori-ressources-ui/node_modules/next/dist/build/index.js:15:924
    at async Span.traceAsyncFn (/Users/julien/code/montessori/montessori-ressources-ui/node_modules/next/dist/telemetry/trace/trace.js:6:584)
error Command failed with exit code 1.
```

When I comment the `target` then it works...

For reference my `package.json`:

```
{
  "name": "next",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "firebase-admin": "^9.10.0",
    "next": "^11.0.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "devDependencies": {
    "autoprefixer": "^10.2.6",
    "eslint": "7.29.0",
    "eslint-config-next": "11.0.1",
    "postcss": "^8.3.5",
    "tailwindcss": "^2.2.4"
  }
}
```
